lua/DCS-gRPC/exporters/object.lua

```diff
@@ -45,6 +45,7 @@ GRPC.exporters.rawTransform = function(object)
 end
 
 GRPC.exporters.group = function(group)
+  if type(group) ~= "table" and getmetatable(group) ~= Group then return {} end

```

Comment:
I think returning `{}` might lead to effects harder to debug than the current error we get. I haven't tested it, but I'd guess this would simply set all fields of the group to their default / undefined value which is probably unexpected in most cases.

Without giving it too much thought (so please feel free to challenge my suggestion), I think the following might be a better solution:
- explicitly define the [`group` property](https://github.com/DCS-gRPC/rust-server/blob/main/protos/dcs/common/v0/common.proto#L235) as `optional`.
- handle the possibility of `getGroup` returning `nil` [in the unit export](https://github.com/DCS-gRPC/rust-server/blob/main/lua/DCS-gRPC/exporters/object.lua#L27)

Wdyt?

Reply:
I initially set it with return nil, but then I suspected that someone might try and iterate over a table. but you are correct. it will force someone to put some guards on his end if.

I'm unfamiliar grpc, but a my gut feeling reading your response is that returning nil in this line, and specifying optional in the proto file is the correct way to go.

I'll make the changes

Reply:
I _really_ don't like marking the group property as optional as we have now complicated the handling of every client (and will probably break existing clients who are expecting group to always be there).

I haven't heard anyone complaining about missing events (that would have these `nil`s) so my question is.

What is the goal of this change? Just stop the error messages appearing? Publishing essential events that are currently not being published?

Reply:
Fair point. I wouldn't mind reducing the DeadEvent down to just containing the unit's name instead of a whole unit object. At least in my use-cases, that is all I need to cleanup state about the units I track in my clients. I'd probably even prefer it over a `UnitWithoutGroup`.

Reply:
Also, I wouldn't be one complaining about missing `Dead` events, as I don't trust ED to fire that reliably anyway. So instead, I just consider a unit dead whenever I fetch information for it and receive a not found error 🤪

Reply:
ED is known to make a mess out for things.. for example, if a unit is dead and burning, it still exists as a staticObject until it's blows up. which is why I assume they disconnect it from the group as staticObjects cannot be assigned to groups.

It's super complicated to replicate but AFAIK you might see something like 
```TRUCK1(Unit in CONVOY) --> TRUCK1(burning staticObject) --> TRUCK1(dead Unit)```

that's super tricky and follow the whole thing I'm afraid. which is sometimes the reason you feel the dead events are unreliable.

Reply:
My goal? Not seeing as many error messages in the log.
There is also a slight chance that once an error pops up dcs event handler will. Not continue processing more functions bound to that event but that might be just a feeling

Reply:
I think the most compatible (But not backwards compatible) solution here is the `DeadUnit` approach.

1. It will get rid of error messages
3. It will give people listening to events the ability to react on this event without needing to consult a local storage of units, which they would otherwise have to do if we just sent the unit ID. This may be important for simple stats trackers.

Reply:
I'm currently trying to debug something on my end. I suspect it has something to do with the grpc throwing LUA errors on parse, but I'll need to be very targeted in my testing, so I'll build a simple test mission tonight and do some tests.

as to how it will be resolved eventually, I trust your judgment for choosing the correct path. but I want to first determine if it's a real issue, as I suspect, or purely cosmetic.

Reply:
Ok, 
I'm circling back to this one. 

I've got a unit testing mission for my project. I had everything green before installing gRPC. 

After compiling from a clean main branch. Tests started failing and the dcs.log was filled with odd nil events from gRPC. Once i re applied this fix, tests are green again. 

Do you have any recommendations about a proper methodology to ensure grpc errors do not break the dcs event handler?